### DISCUSSION TITLE
Adding CTE to query builder

### DISCUSSION THREAD

**Discussion by j3yzz** [laravel/framework#45970] [Ideas]
I believe that adding CTE (Common Table Expression) support to the Query Builder in Laravel v10 is a great idea! CTEs offer powerful capabilities for constructing complex queries, and with this feature, developers can easily join tables, make use of subqueries, and employ unions - all without having to manually write any custom SQL. This will not only help developers write more efficient queries, but it will also make working with complex databases much simpler.

@taylorotwell

**Comment by wesleylether** [comment]
@taylorotwell This would be a great feature for Eloquent. The WITH query is very efficient and can be reused for other WITH's or subqueries 😄

**Reply by macropay-solutions** [reply]
We ended up implementing a similar thing by overwriting the query builder. Basically we create temporary tables for sub-selects to speed up the execution time.
